In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.patches import Rectangle


## DATASET 1: CALIDAD DEL AGUA

In [7]:
pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/waterquality.csv')


aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

In [8]:
df = pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/waterquality.csv')


### SOBRE EL DATASET

CONTEXTO ANALITICO

    El Dataset con el que se trabajará a continuación corresponde a un conjunto de datos referidos al contenido de minerales del agua en un pueblo llamado Dioncaster, el cual se ubica en el extremo occidental del condado de Sabbatown.
    El archivo .csv utilizado fue descargado de la página kaggle.com.

CONTEXTO COMERCIAL

    El objetivo del presente trabajo es analizar como impacta el contenido de determinados minerales en la calidad de agua, la cual es utilizada tanto para riego de cultivos, como para consumo animal. Por lo cual se pretende discriminar en cuanto a la aptitud del agua, en base al fin con el que se la vaya a utilizar.

DESCRIPCIÓN  

Todos los atributos del dataset listados a continuación están expresados por litro:
- aluminium: aluminio
- arsenic: arsénico
- barium: bario
- cadmium: cadmio
- chloramine: cloramina
- chromium: cromo
- cooper: cobre
- fluoride: fluoruro
- bacteria: bacterias
- viruses: virus
- lead: plomo
- nitrates: nitratos
- nitrites: nitritos
- mercury: mercurio
- perchlorate: perclorato
- radium: radio
- selenium: selenio
- silver: plata
- uranium: uranio
- is_safe: es seguro para consumo



El dataset seleccionado cuenta con 7999 filas y 21 columnas

In [10]:
display(df.shape)


(7999, 21)

El dataset cuenta con los siguientes nombres en sus columnas

In [11]:
df.columns

Index(['aluminium', 'ammonia', 'arsenic', 'barium', 'cadmium', 'chloramine',
       'chromium', 'copper', 'flouride', 'bacteria', 'viruses', 'lead',
       'nitrates', 'nitrites', 'mercury', 'perchlorate', 'radium', 'selenium',
       'silver', 'uranium', 'is_safe'],
      dtype='object')

El dataset con los siguientes tipos de datos en sus columnas

In [13]:
df.dtypes

aluminium      float64
ammonia         object
arsenic        float64
barium         float64
cadmium        float64
chloramine     float64
chromium       float64
copper         float64
flouride       float64
bacteria       float64
viruses        float64
lead           float64
nitrates       float64
nitrites       float64
mercury        float64
perchlorate    float64
radium         float64
selenium       float64
silver         float64
uranium        float64
is_safe         object
dtype: object

La información proporcionada es la siguiente:

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7999 entries, 0 to 7998
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   aluminium    7999 non-null   float64
 1   ammonia      7999 non-null   object 
 2   arsenic      7999 non-null   float64
 3   barium       7999 non-null   float64
 4   cadmium      7999 non-null   float64
 5   chloramine   7999 non-null   float64
 6   chromium     7999 non-null   float64
 7   copper       7999 non-null   float64
 8   flouride     7999 non-null   float64
 9   bacteria     7999 non-null   float64
 10  viruses      7999 non-null   float64
 11  lead         7999 non-null   float64
 12  nitrates     7999 non-null   float64
 13  nitrites     7999 non-null   float64
 14  mercury      7999 non-null   float64
 15  perchlorate  7999 non-null   float64
 16  radium       7999 non-null   float64
 17  selenium     7999 non-null   float64
 18  silver       7999 non-null   float64
 19  uraniu

### Las variables que se busca predecir es la llamada "is_safe" a=is_safe 
### Es decir si el agua es apta para consumo animal.


#### Las variables consideradas relevantes para predecir la aptitud en consumo animal son z1=arsenic z2=fluorides z3=nitrates

#### z1=arsenic y z2=fluorides se ha seleccionado ya que el As y el F por si mismos pueden definir la ineptitud del agua para ser consumida. SOn elementos altamente tóxicos para animales y para el hombre, por lo que independientemente de la composición salina del agua en bebida, altos niveles de uno u otro limitan su uso.
#### z3=nitrates se ha elegido ya que los nitratos se consideran peligrosos a partir de 200 ppm y son indicadores de contaminación orgánica, también se ha encontrado que están muy relacionados a la fertilización irracional con urea.